pkg/build/controller/build/build_controller.go

```diff
@@ -1551,27 +1627,39 @@ func (bc *BuildController) createBuildCAConfigMap(build *buildv1.Build, buildPod
 // used by the build pod.
 // The returned ConfigMap has an owner reference to the provided pod, ensuring proper
 // garbage collection.
-func (bc *BuildController) createBuildCAConfigMapSpec(build *buildv1.Build, buildPod *corev1.Pod) *corev1.ConfigMap {
-	// Create a ConfigMap with the `inject-cabundle` annotation, which tells the service cert signing controller
-	// to add the service-ca.crt entry to this map.
-	// See https://github.com/openshift/service-serving-cert-signer
+func (bc *BuildController) createBuildCAConfigMapSpec(build *buildv1.Build, buildPod *corev1.Pod, caData map[string]string) *corev1.ConfigMap {
 	cm := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: buildapihelpers.GetBuildCAConfigMapName(build),
 			OwnerReferences: []metav1.OwnerReference{
 				makeBuildPodOwnerRef(buildPod),
 			},
-			Annotations: map[string]string{
-				"service.alpha.openshift.io/inject-cabundle": "true",
-			},
 		},
-		Data: bc.additionalTrustedCAData,
+		Data: caData,
 	}
 
-	if cm.Data == nil {
-		cm.Data = make(map[string]string)
+	// Fetch the registry's certificate authority stored in the openshift-service-ca ConfigMap
+	registryCAMap, err := bc.controllerManagerConfigMapStore.ConfigMaps("openshift-controller-manager").Get("openshift-service-ca")
+	// If we can't find the CA for the registry, pull from/push to the registry within a build will fail
+	// Logging warnings/errors because the origin/integration tests fail if we raise an error
+	if errors.IsNotFound(err) {
+		klog.V(2).Infof("WARNING - certificate authority for the internal registry could not be found. Image pulls/pushes to the internal registry within build %s/%s will fail.", build.Namespace, build.Name)
+		return cm
```

Comment:
@deads2k another case where the integration test breaks us. I'd rather fail the build here than wait for a `PullBuilderImage` error at execution time.

Reply:
in a followup, make this an actual warning. (klog.Warningf)